### ISSUE TITLE
Change avatarUrl from http to https to avoid networksecurityconfig

### ISSUE THREAD

**Issue by hiimak** [playgameservices/play-games-plugin-for-unity#2563]
It's not possible to retrieve the Player image from the avatarUrl in new Android API versions without a networksecurityconfig in the AndroidManifest.xml. The avatarUrl  defaults to "http://lh3.googleaccounts.com/....". Change to https to clear this bug. 

I have solved this right now by replacing http with https on initialisation.

```csharp

        internal PlayGamesUserProfile(string displayName, string playerId,
            string avatarUrl)
        {
            mDisplayName = displayName;
            mPlayerId = playerId;

            if (avatarUrl.Contains("http"))
            {
                avatarUrl = avatarUrl.Replace("http", "https");
            }
            
            mAvatarUrl = avatarUrl;

            mImageLoading = false;
        }

**Comment by OriginalsByToto** [comment]
You should not check for http, but for http://, because if it's changed to https, it will be matched and will became httpss.
You should also modify the URL in the ResetIdentity function, so you can do a VerifyAvatarURL function and call it from these 2 points:

```private string VerifyAvatarURL(string url)
        {
            return url.Replace("http://", "https://");
        }
```

I've checked the code to find a way to modify it directly in the plugin, but I don't know where is the DLL source, because in GooglePlayGames.Native.Cwrapper.Player the Player_AvatarURL is set to an extern function. Maybe someone knows where it goes...

**Comment by arakhorshid** [comment]
True, thanks for the suggestion, it's the best and shortest solution to the current problem. I searched for the source to, couldn't locate it.

**Comment by vaibhavgpt-google** [comment]
Closing this issue as part of a cleanup of older tickets. Google Play Games v1 is now deprecated. Please upgrade to version v2.1.0 or newer: https://github.com/playgameservices/play-games-plugin-for-unity/releases/tag/v2.1.0. If this issue persists on the new v2 architecture, please open a new ticket.
